store/sqlstore/thread_store.go

```diff
@@ -501,14 +501,21 @@ func (s *SqlThreadStore) DeleteMembershipForUser(userId string, postId string) e
 	return nil
 }
 
-func (s *SqlThreadStore) CreateMembershipIfNeeded(userId, postId string, following, incrementMentions, updateFollowing bool) error {
+func (s *SqlThreadStore) MaintainMembership(userId, postId string, following, incrementMentions, updateFollowing, updateViewedTimestamp bool) error {
 	membership, err := s.GetMembershipForUser(userId, postId)
 	now := utils.MillisFromTime(time.Now())
+	// if memebership exists, update it if:
+	// a. user started/stopped following a thread
+	// b. mention count changed
+	// c. user viewed a thread
 	if err == nil {
-		if (updateFollowing && !membership.Following || membership.Following != following) || incrementMentions {
+		if (updateFollowing && !membership.Following || membership.Following != following) || incrementMentions || updateViewedTimestamp {
 			if updateFollowing {
 				membership.Following = following
 			}
+			if updateViewedTimestamp {
+				membership.LastViewed = now
+			}
```

Comment:
So why are we checking for `membership.Following != following` but then rely solely on `updateFollowing` to do the update? Do we have redundant logic/variables or are we possibly missing a case?